multi-step-form/src/app/add-ons/add-ons.component.ts

```diff
@@ -1,29 +1,62 @@
 import { Component } from '@angular/core';
 import { RouterLink } from '@angular/router';
 import { AddOnsService } from '../add-ons.service';
+import { SelectPlanService } from '../select-plan.service';
 import { CommonModule } from '@angular/common';
 import { FormsModule } from '@angular/forms';
+import { AddOns } from '../add-ons-interface';
 
 @Component({
   selector: 'app-add-ons',
   standalone: true,
   imports: [RouterLink, CommonModule, FormsModule],
   templateUrl: './add-ons.component.html',
-  styleUrl: './add-ons.component.css',
+  styleUrls: ['./add-ons.component.css'],
 })
 export class AddOnsComponent {
-  public monthly = true;
-  public addOnsContainer: any = {};
+  public isToggled = true;
+  public addOnsContainer: AddOns[] = [];
+  public selectedAddOns: AddOns[] = [];
 
-  constructor(readonly addOnsService: AddOnsService) {}
+  constructor(
+    readonly addOnsService: AddOnsService,
+    readonly selectPlanService: SelectPlanService
+  ) {}
 
   ngOnInit(): void {
     this.addOnsService.getAddOnsData().subscribe((data) => {
```

Comment:
move the whole of the subscrption into a private method maybe `initSelectedAddOns`